.github/workflows/snl-ci.yml

```diff
@@ -53,3 +53,38 @@ jobs:
         working-directory: kokkos/build
         run: ctest --output-on-failure --timeout 3600
 
+  HIP_5_6_1:
+    name: SNL_HIP_ROCM_5_6_1
```

Comment:
We can adjust labels later but noting that we are not consistent with the CUDA build naming
(not suggesting that the CUDA build name is much better)